src/Core/Framework/DataAbstractionLayer/Dbal/Common/RepositoryIterator.php

```diff
@@ -71,19 +90,29 @@ public function getTotal(): int
      */
     public function fetchIds(): ?array
     {
+        if ($this->endReached) {
```

Comment:
What about 

```
        
        $repositoryIterator->loop(function(array $entities) {
            
        });
```

In this case we don't have to leak all of this internal "rewind" etc

Reply:
How would you break this loop? Return false? In my suggestions of the getIterator implementation you have the option to just not continue using the iterator.

Reply:
I add some generators as we once written about. This allows for easier flow control by using language features.